src/Handler/Workspace/Create/CreateWorkspaceHandler.php

```diff
@@ -0,0 +1,118 @@
+
+        $db = $this->manager->createSession($credentials);
+        try {
+            // root user is also root database
+            $databaseName = $credentials->getPrincipal();
+
+            // TODO jaky stackPrefix?
+            $nameGenerator = NameGeneratorFactory::getGeneratorForBackendAndPrefix(
+                BackendSupportsInterface::BACKEND_TERADATA,
+                'workspace_'
+            );
+            // TODO pouzit projectId
+            $newWsDatabaseName = $nameGenerator->createWorkspaceObjectNameForWorkspaceId($command->getWorkspaceId());
+            $newWsUserName = $nameGenerator->createWorkspaceUserNameForWorkspaceId($command->getWorkspaceId());
+            $newWsRoleName = $nameGenerator->createWorkspaceRoleNameForWorkspaceId($command->getWorkspaceId());
+            $newWsPassword = Password::generate(
+                30,
+                Password::SET_LOWERCASE | Password::SET_UPPERCASE | Password::SET_NUMBER | Password::SET_SPECIAL_CHARACTERS
+            );
+
+            $db->executeStatement(sprintf(
+                'CREATE ROLE %s;',
+                TeradataQuote::quoteSingleIdentifier($newWsRoleName)
+            ));
+
+            // grant workspace role to root user
+            $db->executeStatement(sprintf(
+                'GRANT %s TO %s;',
+                TeradataQuote::quoteSingleIdentifier($newWsRoleName),
+                TeradataQuote::quoteSingleIdentifier($credentials->getPrincipal()),
+            ));
+
+            // grant workspace role to project user
+            $db->executeStatement(sprintf(
+                'GRANT %s TO %s;',
+                TeradataQuote::quoteSingleIdentifier($newWsRoleName),
+                TeradataQuote::quoteSingleIdentifier($command->getProjectRoleName()),
+            ));
+
+            // grant workspace role to project read only user
+            $db->executeStatement(sprintf(
+                'GRANT SELECT ON %s TO %s;',
+                TeradataQuote::quoteSingleIdentifier($newWsRoleName),
+                TeradataQuote::quoteSingleIdentifier($command->getProjectReadOnlyRoleName()),
+            ));
+
+            // create workspace
+            $db->executeStatement(sprintf(
+                'CREATE USER %s FROM %s AS '
+                . 'PERMANENT = %s, SPOOL = %s, '
+                . 'PASSWORD = %s, DEFAULT DATABASE=%s, DEFAULT ROLE=%s;',
+                TeradataQuote::quoteSingleIdentifier($newWsUserName),
+                TeradataQuote::quoteSingleIdentifier($databaseName),
+                self::DEFAULT_PERM_SPACE_SIZE,
+                self::DEFAULT_SPOOL_SPACE_SIZE,
+                TeradataQuote::quoteSingleIdentifier($newWsPassword),
+                TeradataQuote::quoteSingleIdentifier($newWsUserName),
+                TeradataQuote::quoteSingleIdentifier($newWsRoleName)
+            ));
+        } finally {
+            $db->close();
+        }
```

Comment:
To nemusíš dělat o to sa stará ten session manager

Reply:
pridaval jsem to tam proto, ze kdyz to vyhodilo exception v prubehu, tak to neuzavrelo spojeni - jeste overim...

Reply:
done